index.js

```diff
@@ -25,63 +24,49 @@ function search(buf) {
 	}
 
 	return false;
-}
-
-exports.buffer = function (buf) {
-	return search(buf);
 };
 
-exports.stream = function (cb) {
-	var prevLastByte = new Buffer(1);
-	var searchDone = false;
+exports.buffer = search;
 
-	return through(function (data, enc, cb2) {
-		if (searchDone) {
-			cb(null, data);
-			return;
-		}
+exports.stream = stream => {
+	return new Promise((resolve, reject) => {
+		let prevLastByte = new Buffer(1);
+		let searchDone = false;
 
-		prevLastByte = new Buffer(data[data.length - 1]);
+		stream.on('data', data => {
+			prevLastByte = new Buffer(data[data.length - 1]);
 
-		var res = exports.buffer(Buffer.concat([prevLastByte, data]), true);
+			if (search(Buffer.concat([prevLastByte, data]))) {
+				searchDone = true;
+				stream.destroy();
+			}
+		});
 
-		if (res === true) {
-			searchDone = true;
-			cb(true);
-		}
+		stream.on('error', reject);
 
-		cb2(null, data);
-	}, function (cb2) {
-		if (!searchDone) {
-			cb(false);
-		}
+		stream.on('end', () => {
+			stream.destroy();
```

Comment:
I guess you're right. My idea was that we can safely stop processing because we already know it is a progressive input. But you're right that someone else might also be listening to that same stream and we would break that. That should never ever be the case so we'll just have to process everything untill it is ended and just call resolve as soon as we know it is progressive. Or is it possible to stop listening to the stream without ending it for other listeners?